src/java.base/share/classes/java/lang/runtime/ValueObjectMethods.java

```diff
@@ -1402,4 +1405,47 @@ private static void recursiveChecks(MethodHandle base, int baseCount) {
         throw new IllegalArgumentException("missing leading MethodHandle parameters: " + mt);
     }
 
+    private static boolean isSubstitutableAlt(Object a, Object b) {
+      // This method assumes a and b are not null and their are both instances of the same value class
+      final Unsafe U = UNSAFE;
+      int[] map = U.getFieldMap(a.getClass());
+      int nbNonRef = map[0];
+      for (int i = 0; i < nbNonRef; i++) {
+        int offset = map[i * 2 + 1];
+        int size = map[i * 2 + 2];
+        int nlong = size / 8;
+        for (int j = 0; j < nlong; j++) {
```

Comment:
Are we sure that for all value objects with `size >= 8`, they are always 8-byte aligned?